assets/src/containers/GradeDistribution.js

```diff
@@ -40,28 +77,57 @@ function GradeDistribution (props) {
         <Grid item xs={12} lg={2}>
           <Table className={classes.table} noBorder tableData={[
             [
-              'My Grade', <strong>{gradeData[0].current_user_grade
+              'My grade', <strong>{gradeData[0].current_user_grade
                 ? `${roundToOneDecimcal(gradeData[0].current_user_grade)}%`
                 : 'There are no grades yet for you in this course'}</strong>
             ],
             [
-              'Average Grade',
+              'Average grade',
               <strong>{roundToOneDecimcal(average(grades))}%</strong>
             ],
             [
-              'Median Grade',
+              'Median grade',
               <strong>{roundToOneDecimcal(median(grades))}%</strong>
             ],
-            ['Number of Students', <strong>{gradeData.length}</strong>]
+            ['Number of students', <strong>{gradeData.length}</strong>]
           ]} />
+          {userSettingLoaded
+            ? <> {'Show my grade'} <Checkbox
+              checked={showGrade}
+              onChange={() => {
+                setSettingChanged(true)
+                setShowGrade(!showGrade)
+              }} />
+            </>
+            : <Spinner />}
+          <Snackbar
```

Comment:
Ideally I'd like to extract the `snackbar` and the associated hooks above into its own component - in the future once the pattern of usage in other components gets clearer, I'll try to extract it.